pkg/splunk/enterprise/indexercluster_test.go

```diff
@@ -95,7 +96,58 @@ func TestApplyIndexerCluster(t *testing.T) {
 	splunkDeletionTester(t, revised, deleteFunc)
 }
 
-func getIndexerClusterPodManagerAndClient(method string, mockHandlers []spltest.MockHTTPHandler, replicas int32) (*indexerClusterPodManager, *spltest.MockHTTPClient) {
+func TestGetClusterMasterClient(t *testing.T) {
+	scopedLog := log.WithName("TestGetClusterMasterClient")
+	cr := enterprisev1.IndexerCluster{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "IndexerCluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "stack1",
+			Namespace: "test",
+		},
+		Spec: enterprisev1.IndexerClusterSpec{
+			Replicas: 1,
+			CommonSplunkSpec: enterprisev1.CommonSplunkSpec{
+				ClusterMasterRef: corev1.ObjectReference{
+					Name: "", /* Empty ClusterMasterRef */
+				},
+			},
+		},
+		Status: enterprisev1.IndexerClusterStatus{
+			ClusterMasterPhase: splcommon.PhaseReady,
+		},
+	}
+	cr.Status.IndexerSecretChanged = append(cr.Status.IndexerSecretChanged, true)
+	secrets := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "splunk-master1-indexer-secrets",
+			Namespace: "test",
+		},
+		Data: map[string][]byte{
+			"password": {'1', '2', '3'},
+		},
+	}
+	mockSplunkClient := &spltest.MockHTTPClient{}
+	mgr := &indexerClusterPodManager{
+		log:     scopedLog,
+		cr:      &cr,
+		secrets: secrets,
+		newSplunkClient: func(managementURI, username, password string) *splclient.SplunkClient {
+			c := splclient.NewSplunkClient(managementURI, username, password)
+			c.Client = mockSplunkClient
+			return c
+		},
+	}
+	c := spltest.NewMockClient()
+	mgr.c = c
+	cm := mgr.getClusterMasterClient()
```

Comment:
**Suggestion**: Could we return an error from **getClusterMasterClient** i.e have (*splclient.SplunkClient, error) as the return type and check for the **"Empty cluster master reference"** in err.Error() i.e:

cm, err := mgr.getClusterMasterClient()
if err.Error() != "Empty cluster master reference" {
      t.Errorf("Didn't recognize empty cluster master reference")
}

Reply:
umm yeah we can do that, but its just another way of testing the error condition and I didn't want to add another variable just to check one invalid scenario. Since anyways, we will never have empty `clusterMasterRef` in `getClusterMasterClient` as we would have already checked that in `validateIndexerClusterSpec`.